src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooBuildPlanService.java

```diff
@@ -521,7 +521,6 @@ private DockerConfiguration dockerConfigurationFor(ProgrammingLanguage programmi
      * @return An array of string containing all the configured docker run argument key-value pairs prefixed with two dashes
      */
     private String[] getDefaultDockerRunArguments() {
-        // Bamboo needs all docker run arguments in separate lines
-        return programmingLanguageConfiguration.getDefaultDockerFlags().entrySet().stream().map(entry -> "--" + entry.getKey() + "=" + entry.getValue()).toArray(String[]::new);
+        return programmingLanguageConfiguration.getDefaultDockerFlags().toArray(new String[0]);
```

Comment:
What's the difference between `toArray(String[]::new)` and `toArray(new String[0])`, if any?

Reply:
The previous version was the `toArray` for a `Stream`, so it required a method reference. This is `List::toArray` which takes an actual array.

Reply:
oh, I was not aware that there are two versions of this, expected this to be overloaded. Thanks.